### ISSUE TITLE
softmax accuracy check error

### ISSUE THREAD

**Issue by desertfire** [pytorch/torchdynamo#1239]
Repro:
```
python benchmarks/microbenchmarks/operatorbench.py --dtype=float16 --accuracy-checking=True --max-samples=20 --op=aten._softmax.default --suite=torchbench
```

Error:
```
Iter 16
Accuracy check failed: aten._softmax.default
tensor([[[[-3.2378e-10,  2.8673e-08,  5.0366e-06,  ..., -7.7871e-09,
            2.1710e-09,  7.1896e-10],
          [ 1.4827e-06,  8.3263e-09, -2.3800e-08,  ...,  2.2177e-08,
           -8.1736e-09,  1.4245e-06],
          [ 1.2023e-09,  1.1747e-08, -2.7374e-08,  ...,  1.0114e-09,
            2.0135e-09,  1.4420e-08],
          ...,
          [-7.1020e-09,  2.5417e-08,  2.0416e-08,  ...,  1.5854e-08,
            2.1663e-06,  2.7902e-06],
          [ 2.0017e-08,  2.3114e-08,  1.7526e-07,  ..., -6.9500e-08,
            1.7755e-08, -1.6702e-08],
          [ 2.6268e-08,  8.0399e-09,  8.7504e-10,  ...,  4.0047e-07,
            2.4358e-08, -1.4061e-09]],

         [[-2.1962e-08,  7.0645e-09,  2.0812e-09,  ..., -1.9693e-08,
           -3.2946e-07, -7.7916e-09],
          [-7.4215e-09,  1.9440e-08,  2.4585e-08,  ...,  7.6630e-08,
            4.3219e-08, -1.9238e-09],
          [ 1.9014e-09, -2.7695e-08,  1.6213e-08,  ...,  2.0140e-08,
           -1.1517e-08,  5.3592e-09],
          ...,
          [ 7.2148e-08,  4.6421e-08,  2.9519e-08,  ...,  1.9741e-09,
            1.4164e-09,  2.6043e-08],
          [ 2.5466e-08,  2.0520e-08, -2.0640e-07,  ..., -8.0236e-09,
            1.0118e-09, -2.5868e-08],
          [-2.1795e-08,  5.1126e-09, -1.2734e-08,  ..., -2.6748e-08,
            3.0006e-08,  1.1903e-07]],

         [[ 8.3441e-09, -2.3641e-08, -1.6980e-08,  ...,  1.0604e-08,
            2.7809e-06,  1.9140e-08],
          [-4.6903e-09,  4.6857e-08,  8.7428e-08,  ...,  1.8110e-09,
            1.2279e-08,  4.6042e-08],
          [-4.7963e-06,  4.3375e-09,  1.7020e-06,  ...,  3.1709e-08,
            3.7327e-06, -9.7825e-09],
          ...,
          [-4.4456e-08,  5.8525e-10,  1.5078e-09,  ..., -1.1984e-08,
            4.0318e-09, -1.3300e-08],
          [ 1.6795e-09,  4.6192e-09,  1.5891e-08,  ...,  1.2432e-09,
           -1.4006e-09,  3.2116e-08],
          [ 4.9911e-09, -1.3682e-08, -3.8347e-07,  ..., -1.0263e-06,
            1.5001e-08,  9.1291e-10]],

         ...,

         [[ 4.4471e-08, -3.1199e-08, -1.3630e-06,  ...,  6.3518e-09,
           -1.1442e-08,  4.6109e-09],
          [ 1.4735e-08, -6.9644e-06,  2.4811e-08,  ..., -1.8487e-07,
           -3.1488e-06,  2.7021e-08],
          [-1.3858e-08,  2.3771e-08,  2.5081e-08,  ...,  7.2100e-09,
           -8.3904e-09,  1.1770e-09],
          ...,
          [-2.7249e-08,  6.0321e-09, -2.5163e-08,  ...,  5.4874e-06,
           -1.1912e-06,  1.6391e-07],
          [ 2.6680e-08, -2.2956e-08, -1.6477e-08,  ...,  1.1139e-06,
           -2.0971e-08, -1.3039e-07],
          [-1.0836e-08,  5.9835e-10,  3.5387e-09,  ...,  2.2453e-08,
            8.9011e-07, -1.1222e-06]],

         [[-2.6289e-09, -4.4485e-08,  1.0584e-09,  ...,  1.3935e-07,
            5.7217e-09, -6.4261e-07],
          [-2.2433e-08, -2.2165e-07,  1.6671e-07,  ...,  8.2575e-10,
            1.0466e-07,  1.4537e-08],
          [ 3.5318e-09,  7.3924e-07, -2.1374e-06,  ..., -1.4866e-07,
            4.3900e-07, -2.0591e-09],
          ...,
          [ 5.8068e-10, -7.2317e-07,  1.5725e-06,  ...,  1.5331e-08,
           -1.0838e-07,  3.2975e-08],
          [-2.7172e-08, -2.5002e-08,  5.8740e-09,  ...,  1.9090e-09,
           -7.4762e-07,  1.8440e-08],
          [-1.6228e-07,  1.4070e-09, -5.1319e-09,  ..., -2.1755e-08,
            7.3574e-07,  1.0673e-06]],

         [[-1.0732e-08,  2.8303e-08,  2.3172e-09,  ...,  2.1821e-08,
           -7.6368e-08, -2.6846e-08],
          [ 1.6731e-08,  7.4651e-09,  5.3303e-10,  ...,  4.0006e-09,
            1.8231e-06, -7.8398e-09],
          [ 4.9197e-07,  1.0319e-06, -2.0146e-07,  ...,  7.8471e-09,
           -9.3918e-08, -1.1364e-08],
          ...,
          [ 2.4737e-08,  8.8573e-09, -2.4978e-08,  ..., -4.1700e-07,
           -1.6663e-08,  1.1673e-09],
          [ 1.5632e-08,  1.7200e-09,  3.0542e-09,  ...,  1.1413e-09,
            1.0913e-08,  1.6371e-10],
          [-3.0117e-07,  1.6854e-08, -2.0009e-08,  ...,  1.1854e-08,
            9.6479e-08, -1.2345e-08]]],


        [[[-2.1025e-07,  6.2393e-09,  2.6967e-08,  ...,  9.1534e-09,
           -2.3458e-07,  1.8180e-08],
          [ 9.9159e-10, -3.3091e-08,  3.8603e-07,  ...,  5.0943e-07,
           -1.1316e-07,  1.4807e-08],
          [-5.6520e-08, -1.6767e-08,  1.3043e-08,  ..., -2.4791e-08,
            7.3009e-10,  1.1827e-08],
          ...,
          [-4.9622e-08,  4.1695e-09, -1.5817e-08,  ...,  4.3947e-09,
           -1.0338e-07, -3.1362e-07],
          [ 1.4761e-08,  2.6129e-08, -1.4826e-08,  ..., -1.9185e-06,
           -2.4351e-08,  2.4958e-09],
          [-3.8563e-09, -4.8094e-08, -3.7141e-09,  ..., -1.6319e-08,
           -2.9435e-08,  1.0228e-08]],

         [[ 2.5686e-06,  4.5911e-09, -1.5064e-06,  ...,  2.1828e-08,
           -1.1316e-06,  1.7658e-08],
          [-1.9481e-08,  1.1018e-06,  1.4136e-08,  ...,  1.7736e-08,
            1.7288e-09,  5.4818e-06],
          [ 1.4754e-09,  2.9804e-09, -2.1969e-09,  ..., -1.9727e-08,
           -2.7531e-08, -1.9046e-08],
          ...,
          [ 2.1091e-09, -3.8072e-06,  4.8196e-08,  ..., -4.4441e-09,
           -4.8988e-07, -4.4441e-09],
          [ 8.0327e-08,  2.9594e-08,  3.0370e-08,  ...,  3.0603e-08,
           -2.7181e-08, -3.7922e-08],
          [ 1.4841e-08, -2.3407e-08,  9.8860e-09,  ...,  1.0584e-09,
            1.6775e-08,  8.5466e-09]],

         [[ 5.0699e-08,  9.7371e-09,  5.9456e-06,  ...,  6.7447e-09,
            1.8159e-08, -1.3453e-09],
          [ 5.7276e-07,  4.5367e-07,  7.7686e-09,  ...,  5.1088e-10,
            2.3257e-09,  3.2699e-06],
          [ 1.3728e-06, -2.8018e-08, -1.8813e-06,  ...,  7.6266e-09,
            7.4506e-09,  1.7984e-08],
          ...,
          [-1.9667e-08, -1.6618e-07,  2.8664e-08,  ..., -9.2434e-08,
            2.8085e-09,  1.4042e-08],
          [-3.2131e-06,  4.2681e-08,  9.4186e-10,  ..., -7.4748e-06,
            6.4365e-09,  1.4191e-07],
          [-3.3906e-09, -6.4310e-09,  3.1268e-09,  ...,  6.5718e-10,
           -2.0808e-08, -3.7224e-08]],

         ...,

         [[ 1.2243e-08, -2.1428e-08,  2.4228e-08,  ...,  7.4683e-10,
            3.7339e-09, -8.0599e-09],
          [-2.8500e-08,  2.1275e-08, -1.1817e-08,  ...,  9.3366e-09,
            1.3318e-06,  2.4469e-08],
          [-2.1466e-08, -1.8597e-08,  2.5956e-08,  ...,  1.9782e-09,
            1.8787e-08,  1.4416e-09],
          ...,
          [ 1.3621e-08,  5.0916e-10, -9.5111e-07,  ..., -6.0757e-09,
            4.0876e-09,  1.6559e-08],
          [-2.3904e-08, -4.0419e-07,  8.6390e-09,  ..., -4.1923e-09,
           -3.0291e-07, -1.5479e-06],
          [-9.0513e-09,  2.7852e-08,  3.7198e-09,  ..., -1.0661e-08,
            7.7245e-09, -7.0953e-09]],

         [[-2.8770e-08,  2.8498e-09,  1.5909e-08,  ...,  3.0841e-09,
            1.1949e-06,  3.2547e-09],
          [ 1.0601e-08,  2.2137e-08, -1.3599e-08,  ...,  6.3020e-10,
            1.0659e-08,  1.9595e-06],
          [ 2.3954e-09,  9.7749e-09,  4.8149e-07,  ...,  2.0035e-08,
           -1.6114e-08,  7.7198e-09],
          ...,
          [ 2.4093e-08,  8.0072e-09,  1.2887e-08,  ..., -1.7732e-08,
           -8.9046e-09,  1.0377e-08],
          [-1.6068e-08,  3.0705e-09,  1.6025e-09,  ...,  1.1592e-08,
            7.9640e-10,  2.0555e-09],
          [-1.0035e-08,  1.3638e-08,  1.8879e-08,  ..., -1.0035e-08,
            1.5938e-08, -1.8428e-08]],

         [[ 1.6723e-08, -2.4497e-08,  2.5605e-09,  ...,  4.1153e-08,
            8.6714e-10,  3.9784e-09],
          [ 1.3243e-06, -1.1022e-07, -1.1762e-08,  ...,  8.7361e-09,
           -3.6865e-09, -2.4607e-08],
          [ 3.0009e-09,  2.6993e-08,  2.4415e-09,  ...,  2.1428e-09,
            3.9129e-09, -1.5163e-08],
          ...,
          [ 2.3978e-08,  1.8002e-06, -2.3494e-08,  ..., -1.7590e-07,
           -5.0393e-09,  1.0781e-08],
          [ 1.4389e-07,  1.8033e-07, -3.2773e-06,  ..., -2.0419e-08,
            2.3742e-08, -2.8363e-08],
          [-9.4995e-08,  5.1262e-09,  2.1458e-06,  ...,  1.0082e-08,
           -3.2624e-06, -5.8164e-08]]]], device='cuda:0')
```

**Comment by ngimel** [comment]
Can't repro on master, script runs without error for me.

**Comment by ngimel** [comment]
Ah, nm, I do get an error on iter 10
